### PR TITLE
Add testapp

### PR DESCRIPTION
**Pull request by juanpsenn**
This app would be useful to test interactively the implementation of new providers like TodoPago

### CODE REVIEW

File: testapp/testapp/testmain/models.py [hunk 1]
```diff
@@ -0,0 +1,22 @@
+# Create your models here.
+from decimal import Decimal
+
+from payments import PurchasedItem
+from payments.models import BasePayment
+
+
+class Payment(BasePayment):
+    def get_failure_url(self):
```
Comment (WhyNotHugo) [inline testapp/testapp/testmain/models.py:9]: This should return the URL to a view where a user is redirected after a failed payment (e.g.: `http://localhost:8000/payment-failed`).

File: testapp/testapp/testmain/models.py [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+        return "http://example.com/failure/"
+
+    def get_success_url(self):
+        return "http://example.com/success/"
```
Comment (WhyNotHugo) [inline testapp/testapp/testmain/models.py:13]: Ditto, but for successful payments.

Comment (juanpsenn) [reply testapp/testapp/testmain/models.py:13]: Should I create generic views or leave it customizable from settings? :thinking: 

Maybe a generic view by default would be a simpler approach.

File: testapp/testapp/testmain/views.py [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+    payment = get_object_or_404(get_payment_model(), id=payment_id)
+    try:
+        form = payment.get_form(data=request.POST or None)
+    except RedirectNeeded as redirect_to:
+        return redirect(str(redirect_to))
+    return TemplateResponse(request, "payment.html", {"form": form, "payment": payment})
+
+
+def payment_success(request):
+    raise NotImplementedError()
```
Comment (WhyNotHugo) [inline testapp/testapp/testmain/views.py:24]: `return HttpResponse("Payment succeeded.")`

File: testapp/testapp/testmain/views.py [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+
+
+def payment_failure(request):
+    raise NotImplementedError()
```
Comment (WhyNotHugo) [inline testapp/testapp/testmain/views.py:28]: `return HttpResponse("Payment failed.")`

File: testapp/testapp/urls.py [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+Function views
+    1. Add an import:  from my_app import views
+    2. Add a URL to urlpatterns:  path('', views.home, name='home')
+Class-based views
+    1. Add an import:  from other_app.views import Home
+    2. Add a URL to urlpatterns:  path('', Home.as_view(), name='home')
+Including another URLconf
+    1. Import the include() function: from django.urls import include, path
+    2. Add a URL to urlpatterns:  path('blog/', include('blog.urls'))
+"""
```
Comment (WhyNotHugo) [inline testapp/testapp/urls.py:15]:
```suggestion
```

**Review by WhyNotHugo — CHANGES_REQUESTED** [review]
Looking good.

Can you also add a view for `/` with a button to create a new payment (optionally, pick the total amount, though not really important). This should be enough to complete an entire flow from start to end interactively.

File: testapp/testapp/testmain/views.py [hunk 6]
```diff
@@ -26,3 +29,23 @@ def payment_success(request):
 
 def payment_failure(request):
     return HttpResponse("Payment failed.")
+
+
+def create_test_payment(request):
```
Comment (juanpsenn) [inline testapp/testapp/testmain/views.py:34]: I think that this should do it for now... Maybe redirect() could be optional

File: testapp/testapp/testmain/models.py (hunk 2 above)
Comment (WhyNotHugo) [reply testapp/testapp/testmain/models.py]: A generic view si fine. This is just so we can use this app to manually test providers, so it's just so we can do 100% of the flow via a gui.

File: testapp/testapp/testmain/views.py [hunk 7]
```diff
@@ -0,0 +1,51 @@
...
+    to the payment details view where the get_form() method is called,
+    or a redirect to the providers link is triggered.
+    """
+    payment = get_payment_model()
+    form = TestPaymentForm(
+        initial={"variant": "default", "currency": "USD", "total": 10.0},
+        data=request.POST or None,
+    )
+    if request.method == "POST" and form.is_valid():
+        p = payment.objects.create(description="Product", **form.cleaned_data)
```
Comment (WhyNotHugo) [inline testapp/testapp/testmain/views.py:49]: The `ModelForm` provides an instance already:

```suggestion
        p = form.instance
        p.description = "Product"
        p.save()
```

File: testapp/testapp/testmain/views.py [hunk 8]
```diff
@@ -0,0 +1,51 @@
+# Create your views here.
+
```
Comment (WhyNotHugo) [inline testapp/testapp/testmain/views.py:2]:
```suggestion
```

File: testapp/testapp/testmain/views.py (hunk 6 above)
Comment (WhyNotHugo) [reply testapp/testapp/testmain/views.py:34]: I think it's good as-is.

**Review by WhyNotHugo — APPROVED** [review]
Looks good. Just very minor details left.

File: testapp/testapp/testmain/views.py (hunk 7 above)
Comment (juanpsenn) [reply testapp/testapp/testmain/views.py]: Nice, didn't know that... I also removed the call to get_payment_model()  in https://github.com/jazzband/django-payments/pull/281/commits/ea0af416f9fddcee79245b1eed849341ccdba865

**Review by WhyNotHugo — APPROVED** [review]
Thanks! This'll be perfect for manually testing and debugging providers.

**Pull request merged**